lib/client/user/mediator-topics/has_session.js

```diff
@@ -0,0 +1,32 @@
+var CONSTANTS = require('../../../constants');
+
+/**
+ * Initialsing a subscriber for logging out a user
+ *
+ * @param {object} userEntityTopics
+ * @param {UserClient} userClient - The User Client
+ */
+module.exports = function hasSessionUser(userEntityTopics, userClient) {
+
+  /**
+   *
+   * Handling logging out the current user.
+   *
+   * @param {object} parameters
+   * @param {string/number} parameters.topicUid  - (Optional)  A unique ID to be used to publish completion / error topics.
+   * @returns {*}
+   */
+  return function handleHasSession(parameters) {
+    var self = this;
+    parameters = parameters || {};
+    var hasSessionDoneTopic = userEntityTopics.getTopic(CONSTANTS.TOPICS.HAS_SESSION, CONSTANTS.DONE_PREFIX, parameters.topicUid);
```

Comment:
We do!
It seems these are registered in `lib/client/user/mediator-topcs/index.js` with [`Topics.on`](https://github.com/feedhenry-raincatcher/raincatcher-mediator/blob/master/lib/topics/index.js#L108), this means that if you return something (preferrably a Promise) from the handler it will already respond on the appropriate topics.

It seems `UserClient#hasSession` does just that, so this entire file could be replaced in `mediator-topics/index.js` by
```javascript
userSubscribers.on('has_session', UserClient.hasSession);
```